modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1243 @@
+    lifetime this < x {
+      _enter();
+      _append(x);
+      _leave();
+    }
+
+    /*
+      Returns `true` if this unrolledLinkedList contains an element equal to the
+      value of `x`, and `false` otherwise.
+
+      :arg x: An element to search for.
+      :type x: `eltType`
+
+      :return: `true` if this unrolledLinkedList contains `x`.
+      :rtype: `bool`
+    */
+    proc const contains(x: eltType): bool {
+      var result = false;
+
+      on this {
+        _enter();
+
+        for item in this do
+          if item == x {
+            result = true;
+            break;
+          }
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Returns a reference to the first item in this unrolledLinkedList.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList will cause the 
+        currently running program to halt. If the `--fast` flag is used, no 
+        safety checks will be performed.
+
+      :return: A reference to the first item in this unrolledLinkedList.
+      :rtype: `ref eltType`
+    */
+    proc ref first() ref {
+      _enter();
+
+      if boundsChecking && _size == 0 {
+        _leave();
+        boundsCheckHalt("Called \"unrolledLinkedList.first\" on an empty unrolledLinkedList.");
+      }
+
+      ref result = _getRef(0);
+      _leave();
+
+      return result;
+    }
+
+    /*
+      Returns a reference to the last item in this unrolledLinkedList.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList will cause the 
+        currently running program to halt. If the `--fast` flag is used, no 
+        safety checks will be performed.
+
+      :return: A reference to the last item in this unrolledLinkedList.
+      :rtype: `ref eltType`
+    */
+    proc ref last() ref {
+      _enter();
+
+      if boundsChecking && _size == 0 {
+        _leave();
+        boundsCheckHalt("Called \"unrolledLinkedList.last\" on an empty unrolledLinkedList.");
+      }
+     
+      ref result = _tail!.data[_tail!.size-1];
+      _leave();
+
+      return result;  
+    }
+
+    pragma "no doc"
+    inline proc ref _extendGeneric(collection) {
+      on this {
+        for item in collection {
+          _append(item);
+        }
+      }
+    }
+
+    /*
+      Extend this unrolledLinkedList by appending a copy of each element 
+      contained in a list.
+
+      :arg other: A list containing elements of the same type as those
+        contained in this list.
+      :type other: `list(eltType)`
+    */
+    proc ref extend(other: list(eltType, ?p)) lifetime this < other {
+      on this {
+        _enter();
+        _extendGeneric(other);
+        _leave();
+      }
+    }
+
+    /*
+      Extend this unrolledLinkedList by appending a copy of each element 
+      contained in a unrolledLinkedList.
+
+      :arg other: A unrolledLinkedList containing elements of the same type as 
+      those contained in this unrolledLinkedList.
```

Comment:
I get a successful `make docs` by adding two more spaces before `those` in this comment and in the same place in the next `proc ref extend()`'s comment.

I don't remember having to install anything manually; I just ran `make docs`.

Though the rst file that produces for the function below says

```
   .. method:: proc ref extend(other: eltTypeunrolledLinkedList?p)
```
instead of something like `other: urolledLinkedList(eltType, ?p))`

A similar thing happens for the `List` module's `extend(other: list(eltType, ?p))`, which produces

```
   .. method:: proc ref extend(other: eltTypelist?p)
```

I don't know if there's already an issue open for that.

Reply:
It worked. Thanks